apisix/plugins/proxy-rewrite.lua

```diff
@@ -311,18 +311,18 @@ do
         for i = 1, field_cnt, 2 do
             local val = core.utils.resolve_var(hdr_op.add[i + 1], ctx.var)
             local header = hdr_op.add[i]
-            core.request.add_header(header, val)
+            core.request.add_header(ctx, header, val)
```

Comment:
We may add a case to ensure the warning message won't be generated anymore.

Reply:
Done